src/tools/continuousintegration.rst

```diff
@@ -89,6 +90,132 @@ To do that the option must be enabled:
 
 The EngageSKA cluster located at the Datacenter of Institute of Telecommunication (IT) in Aveiro provides some virtual machines available adding the tag "engageska" or "docker-executor" as shown `here <https://github.com/ska-telescope/ska-python-skeleton/blob/master/.gitlab-ci.yml>`__.
 
+CI pipeline stage descriptions
+------------------------------
+
+.. caution:: This section is a work in progress
+
+The CI/CD pipeline will ensure that software projects are packaged, tested and released in a consistent and predictable manner.
+SKA Pipelines are viewable and executable at https://gitlab.com/ska-telescope
+
+General Notes
+_____________
+- Every commit could potentially trigger a pipeline build. There may be different rules applied to determine which stages are executed in the pipeline based on factors like the branch name.
+
+    - E.g Every commit in a feature branch may trigger the “Lint” stage, but not a slow test suite.
+- When doing a release with a git tag, the full pipeline will be run.
+- Every pipeline job is associated with its git commit (including tag commits).
+- Try and have the stages complete as fast as possible.
+
+    - In some cases it may be possible to parallelize jobs. For example, unit tests and static analysis could be run in parallel.
+- If a job fails, the pipeline will be aborted and the cause of the failure needs to be addressed.
```

Comment:
not necessarily true for all stages. Still you might want to execute other stages and collect metrics without publishing the final product.

Reply:
👍